src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTransformer.java

```diff
@@ -0,0 +1,313 @@
+              conf.getRetrySettingsConfigName(), conf.getRetryCodesConfigName()));
+    }
+
+    TreeMap<RetryConfigDefinitionView.Name, RetryConfigDefinitionView> retryDef = new TreeMap<>();
+    Map<String, ImmutableSet<Code>> retryCodesDef =
+        context.getInterfaceConfig().getRetryCodesDefinition();
+    Map<String, RetrySettings> retryParamsDef =
+        context.getInterfaceConfig().getRetrySettingsDefinition();
+    for (RetryConfigDefinitionView.Name name : retryNames) {
+      ImmutableSet<Code> codes = retryCodesDef.get(name.codes());
+      if (codes.isEmpty()) {
+        continue;
+      }
+      List<String> retryCodeNames = new ArrayList<>();
+      for (Code code : codes) {
+        retryCodeNames.add(namer.getStatusCodeName(code));
+      }
+      retryDef.put(
+          name,
+          RetryConfigDefinitionView.newBuilder()
+              .name(name)
+              .retryCodes(retryCodeNames)
+              .params(retryParamsDef.get(name.params()))
+              .build());
+    }
+    if (!retryDef.isEmpty()) {
+      for (String imp : constant.retryImports) {
+        context.getTypeTable().saveNicknameFor(imp);
+      }
+    }
+    return new ArrayList<>(retryDef.values());
+  }
+
+  private Set<RetryConfigDefinitionView.Name> getRetryNames(SurfaceTransformerContext context) {
+    Set<RetryConfigDefinitionView.Name> retryNames = new HashSet<>();
+    for (Method method : context.getNonStreamingMethods()) {
+      MethodConfig conf = context.getMethodConfig(method);
+      retryNames.add(
+          RetryConfigDefinitionView.Name.create(
+              conf.getRetrySettingsConfigName(), conf.getRetryCodesConfigName()));
+    }
+    return retryNames;
+  }
+
+  private static final String EMPTY_PROTO_PKG = "github.com/golang/protobuf/ptypes/empty";
+
+  private void addImports(SurfaceTransformerContext context, Constant constant) {
+    ModelTypeTable typeTable = context.getTypeTable();
+    for (String imp : constant.unconditionalImports) {
+      typeTable.saveNicknameFor(imp);
+    }
+    typeTable.getImports().remove(EMPTY_PROTO_PKG);
+  }
+
+  private enum Constant {
+    XAPI(
+        XAPI_TEMPLATE_FILENAME,
+        "_client.go",
+        ImmutableList.<String>of(
+            "fmt;;;",
+            "runtime;;;",
+            "golang.org/x/net/context;;;",
+            "google.golang.org/grpc;;;",
+            "google.golang.org/grpc/metadata;;;",
+            "github.com/googleapis/gax-go;gax;;",
+            "google.golang.org/api/option;;;",
+            "google.golang.org/api/transport;;;"),
+        ImmutableList.<String>of("time;;;", "google.golang.org/grpc/codes;;;"),
+        ImmutableList.<String>of("math;;;"),
+        false),
+    SAMPLE(
+        SAMPLE_TEMPLATE_FILENAME,
+        "_client_example_test.go",
+        ImmutableList.<String>of("golang.org/x/net/context;;;"),
+        ImmutableList.<String>of(),
+        ImmutableList.<String>of(),
+        true);
```

Comment:
I'm concerned about this design. It seems like this class is a grab-bag; in the SAMPLE case here there are a bunch of empty inputs, and I can't tell what is what in this particular spot.